sklearn/ensemble/tests/test_iforest.py

```diff
@@ -128,7 +128,19 @@ def test_max_samples_attribute():
     assert_equal(clf.max_samples_, X.shape[0])
 
     clf = IsolationForest(max_samples=0.4).fit(X)
-    assert_equal(clf.max_samples_, 0.4*X.shape[0])
+    assert_equal(clf.max_samples_, 0.4 * X.shape[0])
+
+
+def test_max_depth_attribute():
+    X = iris.data
+    clf = IsolationForest().fit(X)
+    assert_equal(clf.max_depth_, int(np.ceil(np.log2(max(clf.max_samples_, 2)))))
+
+    clf = IsolationForest(max_depth=500).fit(X)
+    assert_equal(clf.max_depth_, clf.max_samples_)
+
+    clf = IsolationForest(max_depth=0.4).fit(X)
+    assert_equal(clf.max_depth_, 0.4 * clf.max_samples_)
 
 
```

Comment:
If I understood it correctly, then we are not allowing users to pass max_depth as a parameter. So we do not need max_depth test cases. Instead we need custom base_estimator test  cases.